src/test/scala/uk/gov/hmcts/ccd/util/CcdTokenGenerator.scala

```diff
@@ -1,27 +1,83 @@
+  val AUTH_CODE_URL = s"$UserAuthUrl/authorizationCode"
+  val AUTH_TOKEN_URL = s"$UserAuthUrl/oauth2/token?client_id=$AUTHORISATION_CLIENT_ID&client_secret=$OAuth2ClientSecret"
+  val restTemplate = new RestTemplate
 
   var dataStoreS2STokenGenerator = applicationContext.getBean("dataStoreS2STokenGenerator").asInstanceOf[AuthTokenGenerator]
   var gatewayS2STokenGenerator = applicationContext.getBean("gatewayS2STokenGenerator").asInstanceOf[AuthTokenGenerator]
 
-  def generateWebUserToken(url: String): String = generateUserToken(UserCcdId, roleFor(url))
+  def generateWebUserToken(): String = generateUserToken(UserCcdId, UserCcdPassword, RedirectUriForAuthToken)
+
+  def generateImportUserToken: String = generateUserToken(UserImportId, UserImportPassword, RedirectUriForAuthToken)
+
+  private def generateUserToken(id: String, password: String, redirectUri: String): String = {
+    val authCode = generateUserAuthCode(id, password, redirectUri)
+    val token = generateUserAuthToken(authCode, redirectUri)
+
+    token
+  }
+
+  private def generateUserAuthCode(userId: String, password: String, redirectUri: String): String = {
+    // retrieve authorisation code for user name and password
+    val headers = new HttpHeaders
+    headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED)
+    headers.setAccept(util.Arrays.asList(MediaType.APPLICATION_JSON_UTF8))
+
+    val formParameters = new LinkedMultiValueMap[String, String]
+    formParameters.add("username", userId)
+    formParameters.add("password", password)
+    formParameters.add("client_id", AUTHORISATION_CLIENT_ID)
+    formParameters.add("redirect_uri", redirectUri)
+
+    val request = new HttpEntity[MultiValueMap[String, String]](formParameters, headers)
+
+    val response = restTemplate.postForEntity(AUTH_CODE_URL, request, classOf[String])
 
-  def generateImportUserToken: String = generateUserToken(UserImportId, "ccd-import")
+    // extract authorisation code from the headers
+    val locationHeader = response.getHeaders.get("Location").toString.replaceAll("\\[", "").replaceAll("\\]", "")
+    val query = new URL(locationHeader).getQuery
+    val queryParameters = parseUrlParameters(query)
 
-  protected def generateUserToken(id: Integer, role: String): String = {
-    val url = TOKEN_LEASE_URL + String.format("?id=%s&role=%s", id, role)
-    val restTemplate = new RestTemplate
-    val response = restTemplate.exchange(url, HttpMethod.POST, null, classOf[String])
-    val token = response.getBody
-    println(s"generated user token $token")
-    "Bearer " + token
+    queryParameters.getOrElse("code", "")
+  }
+
+  private def generateUserAuthToken(authCode: String, redirectUri: String): String = {
+    // retrieve SIDAM authorisation code for user name and password
+    val headers = new HttpHeaders
+    headers.setContentType(MediaType.APPLICATION_FORM_URLENCODED)
+    headers.setAccept(util.Arrays.asList(MediaType.APPLICATION_JSON_UTF8))
```

Comment:
if you import `import scala.collection.JavaConversions._`
you can just pass here a Scala List and that will be automatically converted
```    
headers.setAccept(List(MediaType.APPLICATION_JSON_UTF8))
```

Reply:
done